### PR TITLE
Fixed tabling through solid objects.

### LINKED ISSUE #8084: Acrobats are non-corporeal
**Problem Description**:
Through exploiting tables and flipping, players can get past what should be physical obstacles with ease.

**What did you expect to happen**: 
Doors, Blast Shields, Walls, and Privacy Shutters to stop people.

**What happened instead**:
Doors, Blast Shields, Walls, and Privacy Shutters don't stop people.

**Why is this bad/What are the consequences:**
Allows players to cheese their way into the HoP office, through bolted doors, walls, and other such things.

**Steps to reproduce the problem**:
Have a table under an object (Door, Blast Shield, Wall, Privacy Shutters)
Alt Click the object.
Grab and table a mob onto the table under said object
Grab said mob again
Do the flip emote.

**Possibly related stuff (which gamemode was it? What were you doing at the time? Was
anything else out of the ordinary happening?)**: Gamemode and circumstance aren't relevant as far as I can imagine, I just duplicated while typing this up.
Purpose told me to report this, so I didn't check if it was a dupe or anything.

### PR DESCRIPTION
**Pull request by alex-gh**
**What does this PR do:**
Fixes #8084

**Changelog:**
:cl: uc_guy
fix: You can no longer table people through walls/doors/shutters.
/:cl:

### CODE REVIEW

**Comment by Spartan6** [comment]
Typo in CL

**Comment by alex-gh** [comment]
Fixed, thanks!

File: code/game/objects/structures.dm [hunk 1]
```diff
@@ -64,26 +64,26 @@
 	if(C == user)
 		do_climb(user)
 
+/obj/structure/proc/density_check()
+	for(var/obj/O in range(0, src))
+		if(O.density == 1 && O != src && !istype(O, /obj/machinery/door/window)) //Ignores windoors, as those already block climbing, otherwise a windoor on the opposite side of a table would prevent climbing.
```
Comment (tigercat2000) [inline code/game/objects/structures.dm]: Only need to check `if(O.density`, not `if(O.density == 1`

Comment (tigercat2000) [reply code/game/objects/structures.dm]: Additionally, I can't pull up the docs right now, but I think `orange` will remove `src` from the list automatically.

File: code/game/objects/structures.dm [hunk 2]
```diff
@@ -64,26 +64,26 @@
 	if(C == user)
 		do_climb(user)
 
+/obj/structure/proc/density_check()
+	for(var/obj/O in range(0, src))
+		if(O.density == 1 && O != src && !istype(O, /obj/machinery/door/window)) //Ignores windoors, as those already block climbing, otherwise a windoor on the opposite side of a table would prevent climbing.
+			return O
+	for(var/turf/T in range(0, src))
```
Comment (tigercat2000) [inline code/game/objects/structures.dm]: This doesn't really need to be a `for` loop given that it has a range of zero and only one turf can be on a coordinate space at a time. Just use `get_turf(src)` instead.

File: code/game/objects/structures.dm [hunk 3]
```diff
@@ -64,26 +64,26 @@
 	if(C == user)
 		do_climb(user)
 
+/obj/structure/proc/density_check()
+	for(var/obj/O in range(0, src))
+		if(O.density == 1 && O != src && !istype(O, /obj/machinery/door/window)) //Ignores windoors, as those already block climbing, otherwise a windoor on the opposite side of a table would prevent climbing.
+			return O
+	for(var/turf/T in range(0, src))
+		if(T.density == 1)
```
Comment (tigercat2000) [inline code/game/objects/structures.dm]: `if(T.density)`

**Comment by alex-gh** [comment]
Updated, tested, works.

Thanks for the suggestions!

**Pull request merged**